### PR TITLE
AVRO-3892: [Rust] Support to resolve fixed from bytes and deserialize bytes in deserialize_any

### PR DESCRIPTION
**Pull request by ZENOTME**
## What is the purpose of the change

*(For example: This pull request improves file read performance by buffering data, fixing AVRO-XXXX.)*

fixing AVRO-3892

This PR:
1. support to resolve fixed from bytes
2. support to deserialize bytes, fixed, decimal in deserialize_any

## Verifying this change

1. For resolving fixed from bytes, I have add related test.
2. For deserializing bytes, fixed, decimal in deserialize_any, seems it don't have test for deserialize_any. If it required, we can consider to add test for whole deserialize_any. And actually I have test it in https://github.com/apache/iceberg-rust/pull/82 locally.

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: lang/rust/avro/src/de.rs [hunk 1]
```diff
@@ -350,6 +354,7 @@ impl<'a, 'de> de::Deserializer<'de> for &'a Deserializer<'de> {
             Value::String(ref s) => visitor.visit_bytes(s.as_bytes()),
             Value::Bytes(ref bytes) | Value::Fixed(_, ref bytes) => visitor.visit_bytes(bytes),
             Value::Uuid(ref u) => visitor.visit_bytes(u.as_bytes()),
+            Value::Decimal(ref d) => visitor.visit_bytes(&d.to_vec()?),
```
Comment (martin-g) [inline lang/rust/avro/src/de.rs:357]: It'd be nice to add new unit tests for the changes above.

**Review by martin-g — CHANGES_REQUESTED** [review]

File: lang/rust/avro/src/de.rs [hunk 2]
```diff
@@ -350,6 +354,7 @@ impl<'a, 'de> de::Deserializer<'de> for &'a Deserializer<'de> {
...
             _ => Err(de::Error::custom(format!(
                 "Expected a String|Bytes|Fixed|Uuid, but got {:?}",
```
Comment (martin-g) [inline lang/rust/avro/src/de.rs:359]:
```suggestion
                "Expected a String|Bytes|Fixed|Uuid|Decimal, but got {:?}",
```

File: lang/rust/avro/src/de.rs (hunk 1 above)
Comment (ZENOTME) [reply lang/rust/avro/src/de.rs:357]: Thanks for review! I have added the test for it.

File: lang/rust/avro/src/de.rs [hunk 3]
```diff
@@ -1315,4 +1323,67 @@ mod tests {
...
+                    Ok(Bytes(v.to_vec()))
+                }
+            }
+            deserializer.deserialize_bytes(BytesVisitor)
+        }
+    }
+
+    #[test]
+    fn test_avro_3892_deserialize_bytes_from_decimal() -> TestResult {
+        let expeced_bytes = BigInt::from(123456789).to_signed_bytes_be();
```
Comment (martin-g) [inline lang/rust/avro/src/de.rs:1373]:
```suggestion
        let expected_bytes = BigInt::from(123456789).to_signed_bytes_be();
```

File: lang/rust/avro/src/de.rs [hunk 4]
```diff
@@ -1315,4 +1323,67 @@ mod tests {
...
+        let value = Value::Decimal(Decimal::from(&expeced_bytes));
+        let raw_bytes = from_value::<Bytes>(&value)?;
+        assert_eq!(raw_bytes.0, expeced_bytes);
+        Ok(())
```
Comment (martin-g) [inline lang/rust/avro/src/de.rs:1377]: Please add tests also for the `Value::Bytes` and `Value::Fixed`, and `Value::Union` with them.

File: lang/rust/avro/src/de.rs [hunk 5]
```diff
@@ -1315,4 +1323,67 @@ mod tests {
...
+    }
+
+    #[test]
+    fn test_avro_3892_deserialize_bytes_from_uuid() -> TestResult {
+        let uuid_str = "10101010-2020-2020-2020-101010101010";
+        let expected_bytes = Uuid::parse_str(uuid_str)?.as_bytes().to_vec();
```
[the remark this answers is not in the document]
Comment (martin-g) [reply lang/rust/avro/src/de.rs:1383]: Dunno!
I have to check what `Uuid::as_bytes()` does. I'll check tomorrow!
But feel free to propose a PR with a fix if you think it is wrong!

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you, @ZENOTME !

**Pull request merged**